packages/hub/routes/email-card-drop-requests.ts

```diff
@@ -222,10 +222,21 @@ export default class EmailCardDropRequestsRoute {
       email,
     });
 
-    await this.workerClient.addJob('subscribe-email', {
-      id: emailCardDropRequest.id,
-      email,
-    });
+    await this.workerClient.addJob(
+      'subscribe-email',
+      { email },
+      // for the most part, Mailchimp errors will not be retriable
+      // because of that we limit maxAttempts to 1
+      // this prevents us from retrying where we cannot do anything
+      // and also leaves dead jobs in the db that we can check later
+      // this should be supplemented by the Mailchimp errors being sent
+      // to Sentry and having appropriate alerts, especially for rate limiting
+      // could consider permanently failing the job from within conditionally, instead. probably as a utility function?
+      // https://github.com/graphile/worker/blob/e3176eab42ada8f4f3718192bada776c22946583/sql/000004.sql#L122-L126
+      {
+        maxAttempts: 1,
```

Comment:
Basically we want to permanently fail whenever we get one of the errors here:

https://mailchimp.com/developer/marketing/docs/errors/#error-glossary

Since it seems unlikely that we will be able to send properly then. These error codes should also trigger Sentry alerts.

Reply:
If you want retry sometimes but not always, marking as permanently failed inside the job seems good. This appears to be the relevant JS function: https://github.com/graphile/worker/blob/e3176eab42ada8f4f3718192bada776c22946583/src/workerUtils.ts#L40

Reply:
I tried using this JS function but it doesn't appear to update the job's state properly. I'm not sure if there is something preventing the permanent failure of jobs using the JS SDK.